src/python/pants/backend/python/util_rules/pex_test.py

```diff
@@ -538,77 +544,153 @@ def assert_description(
     )
 
 
+DEFAULT = "DEFAULT"
+OTHER_CONTENT = "OTHER_CONTENT"
+FILE = "FILE"
+
+
 def test_error_on_invalid_lockfile_with_path(rule_runner: RuleRunner) -> None:
     with pytest.raises(ExecutionError):
         _run_pex_for_lockfile_test(
             rule_runner,
-            use_file=True,
+            lockfile_type=FILE,
             behavior="error",
             invalid_reqs=True,
         )
 
 
-def test_warn_on_invalid_lockfile_with_path(rule_runner: RuleRunner, caplog) -> None:
-    _run_pex_for_lockfile_test(rule_runner, use_file=True, behavior="warn", invalid_reqs=True)
-    assert "Invalid lockfile for PEX request" in caplog.text
+def test_ignore_on_invalid_lockfile_with_path(rule_runner: RuleRunner, caplog) -> None:
+    _run_pex_for_lockfile_test(
+        rule_runner, lockfile_type=FILE, behavior="ignore", invalid_reqs=True
+    )
+    assert not caplog.text.strip()
 
 
-def test_warn_on_requirements_mismatch(rule_runner: RuleRunner, caplog) -> None:
-    _run_pex_for_lockfile_test(rule_runner, use_file=True, behavior="warn", invalid_reqs=True)
-    assert "requirements set for this" in caplog.text
-    assert "generated under interpreter constraints" not in caplog.text
+def test_no_warning_on_valid_lockfile_with_path(rule_runner: RuleRunner, caplog) -> None:
+    _run_pex_for_lockfile_test(rule_runner, lockfile_type=FILE, behavior="warn")
+    assert not caplog.text.strip()
 
 
-def test_warn_on_interpreter_constraints_mismatch(rule_runner: RuleRunner, caplog) -> None:
-    _run_pex_for_lockfile_test(
-        rule_runner, use_file=True, behavior="warn", invalid_constraints=True
-    )
-    assert "requirements set for this" not in caplog.text
-    assert "generated under interpreter constraints" in caplog.text
+def test_error_on_invalid_lockfile_with_content(rule_runner: RuleRunner) -> None:
+    with pytest.raises(ExecutionError):
+        _run_pex_for_lockfile_test(
+            rule_runner, lockfile_type=OTHER_CONTENT, behavior="error", invalid_reqs=True
+        )
+
+
+def test_no_warning_on_valid_lockfile_with_content(rule_runner: RuleRunner, caplog) -> None:
+    _run_pex_for_lockfile_test(rule_runner, lockfile_type=OTHER_CONTENT, behavior="warn")
+    assert not caplog.text.strip()
+
+
+LOCKFILE_TYPES = (DEFAULT, OTHER_CONTENT, FILE)
+BOOLEANS = (True, False)
 
 
-def test_warn_on_mismatched_requirements_and_interpreter_constraints(
-    rule_runner: RuleRunner, caplog
+@pytest.mark.parametrize(
+    "lockfile_type,invalid_reqs,invalid_constraints,uses_source_plugins,uses_project_ic",
+    [
+        (lft, ir, ic, usp, upi)
+        for lft in LOCKFILE_TYPES
+        for ir in BOOLEANS
+        for ic in BOOLEANS
+        for usp in BOOLEANS
+        for upi in BOOLEANS
+    ],
+)
+def test_lockfile_warnings(
```

Comment:
As mentioned on DM, most these tests should never build a PEX and only test your helper function directly.

Reply:
Yup, working on that presently.